### ISSUE TITLE
Spyder 2.3.8 vs. Matplotlib 1.5 vs. latex fonts

### ISSUE THREAD

**Issue by bs538** [spyder-ide/spyder#2859]
In the Spyder (v. 2.3.8) IPython Console I get a runtime error from matplotlib ("RuntimeError: In select_charmap: Could not set the charmap") when rendering plots with Latex fonts. To enable the Latex fonts, I use these settings:

    plt.rc('text', usetex=True)
    plt.rc('font', **{'family':'serif', 'serif':['Times'], 'size':24})

In a normal IPython Console the same plot with Latex labels renders fine, so I suspect the issue is related to the default graphics backend in the Spyder IPython console.

Disabling the latex fonts or downgrading to Matplotlib 1.4.3 both make the problem disappear. Below I have copied the full error traceback.

```
  File "/data/bs538/installed/anaconda/lib/python2.7/site-packages/IPython/core/formatters.py", line 337, in __call__
    return printer(obj)

  File "/data/bs538/installed/anaconda/lib/python2.7/site-packages/IPython/core/pylabtools.py", line 213, in <lambda>
    svg_formatter.for_type(Figure, lambda fig: print_figure(fig, 'svg', **kwargs))

  File "/data/bs538/installed/anaconda/lib/python2.7/site-packages/IPython/core/pylabtools.py", line 117, in print_figure
    fig.canvas.print_figure(bytes_io, **kw)

  File "/data/bs538/installed/anaconda/lib/python2.7/site-packages/matplotlib/backend_bases.py", line 2178, in print_figure
    **kwargs)

  File "/data/bs538/installed/anaconda/lib/python2.7/site-packages/matplotlib/backends/backend_svg.py", line 1206, in print_svg
    return self._print_svg(filename, svgwriter, fh_to_close, **kwargs)

  File "/data/bs538/installed/anaconda/lib/python2.7/site-packages/matplotlib/backends/backend_svg.py", line 1234, in _print_svg
    self.figure.draw(renderer)

  File "/data/bs538/installed/anaconda/lib/python2.7/site-packages/matplotlib/artist.py", line 61, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)

  File "/data/bs538/installed/anaconda/lib/python2.7/site-packages/matplotlib/figure.py", line 1133, in draw
    func(*args)

  File "/data/bs538/installed/anaconda/lib/python2.7/site-packages/matplotlib/artist.py", line 61, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)

  File "/data/bs538/installed/anaconda/lib/python2.7/site-packages/matplotlib/axes/_base.py", line 2304, in draw
    a.draw(renderer)

  File "/data/bs538/installed/anaconda/lib/python2.7/site-packages/matplotlib/artist.py", line 61, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)

  File "/data/bs538/installed/anaconda/lib/python2.7/site-packages/matplotlib/axis.py", line 1111, in draw
    tick.draw(renderer)

  File "/data/bs538/installed/anaconda/lib/python2.7/site-packages/matplotlib/artist.py", line 61, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)

  File "/data/bs538/installed/anaconda/lib/python2.7/site-packages/matplotlib/axis.py", line 252, in draw
    self.label1.draw(renderer)

  File "/data/bs538/installed/anaconda/lib/python2.7/site-packages/matplotlib/artist.py", line 61, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)

  File "/data/bs538/installed/anaconda/lib/python2.7/site-packages/matplotlib/text.py", line 792, in draw
    mtext=mtext)

  File "/data/bs538/installed/anaconda/lib/python2.7/site-packages/matplotlib/backends/backend_svg.py", line 1152, in draw_tex
    self._draw_text_as_path(gc, x, y, s, prop, angle, ismath="TeX")

  File "/data/bs538/installed/anaconda/lib/python2.7/site-packages/matplotlib/backends/backend_svg.py", line 957, in _draw_text_as_path
    return_new_glyphs_only=True)

  File "/data/bs538/installed/anaconda/lib/python2.7/site-packages/matplotlib/textpath.py", line 346, in get_glyphs_tex
    font.select_charmap(charmap_code)

RuntimeError: In select_charmap: Could not set the charmap
```

**Comment by ccordoba12** [comment]
Are you sure this is a Spyder bug instead of a Matplotlib one?

I mean, what happens if you run the same code in terminal IPython or in qtconsole?

**Comment by bs538** [comment]
Thanks for the quick response. The same code runs fine and produces a figure in a standard IPython console.

**Comment by ccordoba12** [comment]
Ok, a couple of questions: What's your

1. Operating system?
2. Python version?
3. Matplotlib backend?

**Comment by bs538** [comment]
Sorry, could have included this information in the first place. 
1. OS: Red Hat Enterprise Linux Workstation release 6.7 (Santiago)
2. Python version: Python 2.7.10 | Anaconda 2.2.0 (64-bit)| IPython 4.0.0
3. Matplotlib backend: Inline (Spyder default)

**Comment by ccordoba12** [comment]
Ok, good. So it's a problem with the SVG backend. i still think this is an issue with Matplotlib instead of Spyder.

Could you post a fully reproducible example to test on my end?

**Comment by bs538** [comment]
Minimal example:

    import matplotlib.pyplot as plt
    import numpy as np

    plt.rc('text', usetex=True)
    plt.rc('font', **{'family':'serif', 'serif':['Times'], 'size':24})

    x=np.linspace(-5,5,100)
    plt.plot(x,np.sin(x))
    plt.savefig('test.png')
    plt.show()

Creating and saving the figure works fine, it is displaying it with plt.show() that creates the error.

**Comment by aginiewicz** [comment]
Seems it might be already fixed in matplotlib 1.5.1: https://github.com/matplotlib/matplotlib/pull/5770#issuecomment-168212746

**Comment by ccordoba12** [comment]
Great! Thanks for letting us know about it :-)
